{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import os \n",
    "from amas import AMAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'gene0': [], 'gene1': [], 'gene2': [], 'gene3': [], 'gene4': [], 'gene5': []}\n"
     ]
    }
   ],
   "source": [
    "all_genes = {}\n",
    "#creates list of all genes \n",
    "for file in os.listdir(\"/Users/sarahharrison/Documents/GitHub/CF-BM-WGS-data-analysis/reference_data/muscle_test/gene_TEST\"):\n",
    "    if file.endswith(\".fasta\") == True:\n",
    "        for seq_record in SeqIO.parse(file, \"fasta\"):\n",
    "            gene_name = seq_record.id.split('_')[0]\n",
    "            if gene_name not in all_genes:\n",
    "                all_genes[gene_name] = []\n",
    "                \n",
    "print(all_genes)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'gene0': [SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_003', name='gene0_003', description='gene0_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_001', name='gene0_001', description='gene0_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_005', name='gene0_005', description='gene0_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_000', name='gene0_000', description='gene0_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_002', name='gene0_002', description='gene0_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_004', name='gene0_004', description='gene0_004', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_003', name='gene0_003', description='gene0_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_001', name='gene0_001', description='gene0_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_005', name='gene0_005', description='gene0_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_000', name='gene0_000', description='gene0_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_002', name='gene0_002', description='gene0_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGCGCAAGTCAATCAAGTCGCGCGAGCGATTGCAGTTCGCGCTCAAGCCGAGT...TGA', SingleLetterAlphabet()), id='gene0_004', name='gene0_004', description='gene0_004', dbxrefs=[])], 'gene1': [SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_003', name='gene1_003', description='gene1_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_001', name='gene1_001', description='gene1_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_005', name='gene1_005', description='gene1_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_000', name='gene1_000', description='gene1_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_002', name='gene1_002', description='gene1_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_004', name='gene1_004', description='gene1_004', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_003', name='gene1_003', description='gene1_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_001', name='gene1_001', description='gene1_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_005', name='gene1_005', description='gene1_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_000', name='gene1_000', description='gene1_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_002', name='gene1_002', description='gene1_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGATTCGATCGGTGCAACGCAGACACATCGTCCGTTGGCTGTCCGCAATATTG...TGA', SingleLetterAlphabet()), id='gene1_004', name='gene1_004', description='gene1_004', dbxrefs=[])], 'gene2': [SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_003', name='gene2_003', description='gene2_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_001', name='gene2_001', description='gene2_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_005', name='gene2_005', description='gene2_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_000', name='gene2_000', description='gene2_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_002', name='gene2_002', description='gene2_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_004', name='gene2_004', description='gene2_004', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_003', name='gene2_003', description='gene2_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_001', name='gene2_001', description='gene2_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_005', name='gene2_005', description='gene2_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_000', name='gene2_000', description='gene2_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_002', name='gene2_002', description='gene2_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGAAAGCAACTCTTCGCTCGCTCGTCGCCGTGGCGGCATTCGCGTCGCTCGCA...TAA', SingleLetterAlphabet()), id='gene2_004', name='gene2_004', description='gene2_004', dbxrefs=[])], 'gene3': [SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_003', name='gene3_003', description='gene3_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_001', name='gene3_001', description='gene3_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_005', name='gene3_005', description='gene3_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_000', name='gene3_000', description='gene3_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_002', name='gene3_002', description='gene3_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_004', name='gene3_004', description='gene3_004', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_003', name='gene3_003', description='gene3_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_001', name='gene3_001', description='gene3_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_005', name='gene3_005', description='gene3_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_000', name='gene3_000', description='gene3_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_002', name='gene3_002', description='gene3_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGCCGGGCCGGCCGCCCGTACCGAGATCGGTCTGCTCCGCGCTGATCGCCGCG...TGA', SingleLetterAlphabet()), id='gene3_004', name='gene3_004', description='gene3_004', dbxrefs=[])], 'gene4': [SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_003', name='gene4_003', description='gene4_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_001', name='gene4_001', description='gene4_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_005', name='gene4_005', description='gene4_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_000', name='gene4_000', description='gene4_000 4', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_002', name='gene4_002', description='gene4_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_004', name='gene4_004', description='gene4_004', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_003', name='gene4_003', description='gene4_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_001', name='gene4_001', description='gene4_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_005', name='gene4_005', description='gene4_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_000', name='gene4_000', description='gene4_000 4', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_002', name='gene4_002', description='gene4_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGGAATCGATCGGCATCGTCGCTCCACAGACCATGCACTTCGCCGAACCGCTG...TGA', SingleLetterAlphabet()), id='gene4_004', name='gene4_004', description='gene4_004', dbxrefs=[])], 'gene5': [SeqRecord(seq=Seq('ATGATCCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_003', name='gene5_003', description='gene5_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGAACCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_001', name='gene5_001', description='gene5_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGATCCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_005', name='gene5_005', description='gene5_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGAACCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_000', name='gene5_000', description='gene5_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGAACCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_002', name='gene5_002', description='gene5_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGATCCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_004', name='gene5_004', description='gene5_004', dbxrefs=[]), SeqRecord(seq=Seq('ATGATCCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_003', name='gene5_003', description='gene5_003', dbxrefs=[]), SeqRecord(seq=Seq('ATGAACCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_001', name='gene5_001', description='gene5_001', dbxrefs=[]), SeqRecord(seq=Seq('ATGATCCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_005', name='gene5_005', description='gene5_005', dbxrefs=[]), SeqRecord(seq=Seq('ATGAACCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_000', name='gene5_000', description='gene5_000', dbxrefs=[]), SeqRecord(seq=Seq('ATGAACCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_002', name='gene5_002', description='gene5_002', dbxrefs=[]), SeqRecord(seq=Seq('ATGATCCAGCAGGCGCTGAACTCCCTGTCCGCGCGCGCGGACTTCCGCGCGATC...TGA', SingleLetterAlphabet()), id='gene5_004', name='gene5_004', description='gene5_004', dbxrefs=[])]}\n"
     ]
    }
   ],
   "source": [
    "for file in os.listdir(\"/Users/sarahharrison/Documents/GitHub/CF-BM-WGS-data-analysis/reference_data/muscle_test/gene_TEST\"):\n",
    "    if file.endswith(\".fasta\") == True:\n",
    "        for seq_record in SeqIO.parse(file, \"fasta\"):\n",
    "            name = seq_record.id.split('_')[0]\n",
    "            all_genes[name].append(seq_record)\n",
    "\n",
    "print(all_genes)\n",
    "for k,v in all_genes.items():\n",
    "    SeqIO.write(v, k + \"gene.fasta\", \"fasta\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "000.fasta       003.fasta       Untitled.ipynb  gene2gene.fasta gene5gene.fasta\r\n",
      "001.fasta       004.fasta       gene0gene.fasta gene3gene.fasta\r\n",
      "002.fasta       005.fasta       gene1gene.fasta gene4gene.fasta\r\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "gene0gene 12 seqs, lengths min 2724, max 2724, avg 2724\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n",
      "00:00:00      3 MB(0%)  Iter   1    9.09%  Align node       \r",
      "00:00:00     17 MB(0%)  Iter   1   18.18%  Align node\r",
      "00:00:00     19 MB(0%)  Iter   1   27.27%  Align node\r",
      "00:00:00     21 MB(0%)  Iter   1   36.36%  Align node\r",
      "00:00:01     22 MB(0%)  Iter   1   45.45%  Align node\r",
      "00:00:01     23 MB(0%)  Iter   1   54.55%  Align node\r",
      "00:00:01     25 MB(0%)  Iter   1   63.64%  Align node\r",
      "00:00:01     28 MB(0%)  Iter   1   72.73%  Align node\r",
      "00:00:01     29 MB(0%)  Iter   1   81.82%  Align node\r",
      "00:00:01     29 MB(0%)  Iter   1   90.91%  Align node\r",
      "00:00:01     30 MB(0%)  Iter   1  100.00%  Align node\r",
      "00:00:01     31 MB(0%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:01     31 MB(0%)  Iter   1    8.33%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   16.67%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   25.00%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   33.33%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   41.67%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   50.00%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   58.33%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   66.67%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   75.00%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   83.33%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1   91.67%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "00:00:01     31 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01     31 MB(0%)  Iter   2   10.00%  Refine tree   \r",
      "00:00:01     32 MB(0%)  Iter   2   20.00%  Refine tree\r",
      "00:00:01     32 MB(0%)  Iter   2   30.00%  Refine tree\r",
      "00:00:01     32 MB(0%)  Iter   2  100.00%  Refine tree\r",
      "\r\n",
      "00:00:01     32 MB(0%)  Iter   2    8.33%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   16.67%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   25.00%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   33.33%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   41.67%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   50.00%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   58.33%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   66.67%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   75.00%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   83.33%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2   91.67%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "00:00:01     32 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01     32 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01     32 MB(0%)  Iter   3    9.52%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   14.29%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   19.05%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   23.81%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   28.57%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   33.33%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   38.10%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   42.86%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   47.62%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   52.38%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   57.14%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   61.90%  Refine biparts\r",
      "00:00:02     32 MB(0%)  Iter   3   66.67%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3   71.43%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3   76.19%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3   80.95%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3   85.71%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3   90.48%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3   95.24%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3  104.76%  Refine biparts\r",
      "00:00:03     32 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "\r\n",
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "gene1gene 12 seqs, lengths min 1041, max 1041, avg 1041\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n",
      "00:00:00      2 MB(0%)  Iter   1    9.09%  Align node       \r",
      "00:00:00      5 MB(0%)  Iter   1   18.18%  Align node\r",
      "00:00:00      6 MB(0%)  Iter   1   27.27%  Align node\r",
      "00:00:00      6 MB(0%)  Iter   1   36.36%  Align node\r",
      "00:00:00      7 MB(0%)  Iter   1   45.45%  Align node\r",
      "00:00:00      8 MB(0%)  Iter   1   54.55%  Align node\r",
      "00:00:00      8 MB(0%)  Iter   1   63.64%  Align node\r",
      "00:00:00      9 MB(0%)  Iter   1   72.73%  Align node\r",
      "00:00:00     10 MB(0%)  Iter   1   81.82%  Align node\r",
      "00:00:00     10 MB(0%)  Iter   1   90.91%  Align node\r",
      "00:00:00     11 MB(0%)  Iter   1  100.00%  Align node\r",
      "00:00:00     11 MB(0%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:00     11 MB(0%)  Iter   1    8.33%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   16.67%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   25.00%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   33.33%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   41.67%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   50.00%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   58.33%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   66.67%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   75.00%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   83.33%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1   91.67%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "00:00:00     11 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00     11 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00     12 MB(0%)  Iter   3    9.52%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   14.29%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   19.05%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   23.81%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   28.57%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   33.33%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   38.10%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   42.86%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   47.62%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   52.38%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   57.14%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   61.90%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   66.67%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   71.43%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   76.19%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   80.95%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   85.71%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   90.48%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3   95.24%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3  104.76%  Refine biparts\r",
      "00:00:00     12 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "\r\n",
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "gene2gene 12 seqs, lengths min 540, max 540, avg 540\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n",
      "00:00:00      2 MB(0%)  Iter   1    9.09%  Align node       \r",
      "00:00:00      3 MB(0%)  Iter   1   18.18%  Align node\r",
      "00:00:00      4 MB(0%)  Iter   1   27.27%  Align node\r",
      "00:00:00      4 MB(0%)  Iter   1   36.36%  Align node\r",
      "00:00:01      4 MB(0%)  Iter   1   45.45%  Align node\r",
      "00:00:01      5 MB(0%)  Iter   1   54.55%  Align node\r",
      "00:00:01      5 MB(0%)  Iter   1   63.64%  Align node\r",
      "00:00:01      5 MB(0%)  Iter   1   72.73%  Align node\r",
      "00:00:01      6 MB(0%)  Iter   1   81.82%  Align node\r",
      "00:00:01      6 MB(0%)  Iter   1   90.91%  Align node\r",
      "00:00:01      6 MB(0%)  Iter   1  100.00%  Align node\r",
      "00:00:01      6 MB(0%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:01      6 MB(0%)  Iter   1    8.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   16.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   25.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   33.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   41.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   50.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   58.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   66.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   75.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   83.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1   91.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01      6 MB(0%)  Iter   2   10.00%  Refine tree   \r",
      "00:00:01      6 MB(0%)  Iter   2   20.00%  Refine tree\r",
      "00:00:01      6 MB(0%)  Iter   2   30.00%  Refine tree\r",
      "00:00:01      6 MB(0%)  Iter   2  100.00%  Refine tree\r",
      "\r\n",
      "00:00:01      6 MB(0%)  Iter   2    8.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   16.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   25.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   33.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   41.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   50.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   58.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   66.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   75.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   83.33%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2   91.67%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "00:00:01      6 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01      6 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01      6 MB(0%)  Iter   3    9.52%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   14.29%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   19.05%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   23.81%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   28.57%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   33.33%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   38.10%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   42.86%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   47.62%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   52.38%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   57.14%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   61.90%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   66.67%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   71.43%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   76.19%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   80.95%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   85.71%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   90.48%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3   95.24%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3  104.76%  Refine biparts\r",
      "00:00:01      6 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "\r\n",
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "gene3gene 12 seqs, lengths min 2991, max 2997, avg 2993\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n",
      "00:00:00      3 MB(0%)  Iter   1    9.09%  Align node       \r",
      "00:00:00     18 MB(0%)  Iter   1   18.18%  Align node\r",
      "00:00:00     21 MB(0%)  Iter   1   27.27%  Align node\r",
      "00:00:00     23 MB(0%)  Iter   1   36.36%  Align node\r",
      "00:00:00     25 MB(0%)  Iter   1   45.45%  Align node\r",
      "00:00:00     27 MB(0%)  Iter   1   54.55%  Align node\r",
      "00:00:00     30 MB(0%)  Iter   1   63.64%  Align node\r",
      "00:00:00     30 MB(0%)  Iter   1   72.73%  Align node\r",
      "00:00:00     31 MB(0%)  Iter   1   81.82%  Align node\r",
      "00:00:01     32 MB(0%)  Iter   1   90.91%  Align node\r",
      "00:00:01     33 MB(0%)  Iter   1  100.00%  Align node\r",
      "00:00:01     34 MB(0%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:01     34 MB(0%)  Iter   1    8.33%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   16.67%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   25.00%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   33.33%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   41.67%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   50.00%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   58.33%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   66.67%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   75.00%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   83.33%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1   91.67%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "00:00:01     34 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "\r\n",
      "00:00:01     34 MB(0%)  Iter   2   10.00%  Refine tree   \r",
      "00:00:01     35 MB(0%)  Iter   2   20.00%  Refine tree\r",
      "00:00:01     35 MB(0%)  Iter   2   30.00%  Refine tree\r",
      "00:00:01     35 MB(0%)  Iter   2   40.00%  Refine tree\r",
      "00:00:01     35 MB(0%)  Iter   2   50.00%  Refine tree\r",
      "00:00:01     35 MB(0%)  Iter   2   60.00%  Refine tree\r",
      "00:00:01     35 MB(0%)  Iter   2   70.00%  Refine tree\r",
      "00:00:02     35 MB(0%)  Iter   2   80.00%  Refine tree\r",
      "00:00:02     35 MB(0%)  Iter   2  100.00%  Refine tree\r",
      "\r\n",
      "00:00:02     35 MB(0%)  Iter   2    8.33%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   16.67%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   25.00%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   33.33%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   41.67%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   50.00%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   58.33%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   66.67%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   75.00%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   83.33%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2   91.67%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "00:00:02     35 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:02     35 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:02     35 MB(0%)  Iter   3    9.52%  Refine biparts\r",
      "00:00:02     35 MB(0%)  Iter   3   14.29%  Refine biparts\r",
      "00:00:02     35 MB(0%)  Iter   3   19.05%  Refine biparts\r",
      "00:00:02     35 MB(0%)  Iter   3   23.81%  Refine biparts\r",
      "00:00:02     35 MB(0%)  Iter   3   28.57%  Refine biparts\r",
      "00:00:02     35 MB(0%)  Iter   3   33.33%  Refine biparts\r",
      "00:00:02     35 MB(0%)  Iter   3   38.10%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   42.86%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   47.62%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   52.38%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   57.14%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   61.90%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   66.67%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   71.43%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   76.19%  Refine biparts\r",
      "00:00:03     35 MB(0%)  Iter   3   80.95%  Refine biparts\r",
      "00:00:04     35 MB(0%)  Iter   3   85.71%  Refine biparts\r",
      "00:00:04     35 MB(0%)  Iter   3   90.48%  Refine biparts\r",
      "00:00:04     35 MB(0%)  Iter   3   95.24%  Refine biparts\r",
      "00:00:04     35 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "00:00:04     35 MB(0%)  Iter   3  104.76%  Refine biparts\r",
      "00:00:04     35 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "\r\n",
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "gene4gene 12 seqs, lengths min 1145, max 1146, avg 1145\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n",
      "00:00:00      2 MB(0%)  Iter   1    9.09%  Align node       \r",
      "00:00:00      6 MB(0%)  Iter   1   18.18%  Align node\r",
      "00:00:00      6 MB(0%)  Iter   1   27.27%  Align node\r",
      "00:00:00      7 MB(0%)  Iter   1   36.36%  Align node\r",
      "00:00:00      8 MB(0%)  Iter   1   45.45%  Align node\r",
      "00:00:00      9 MB(0%)  Iter   1   54.55%  Align node\r",
      "00:00:00      9 MB(0%)  Iter   1   63.64%  Align node\r",
      "00:00:00     10 MB(0%)  Iter   1   72.73%  Align node\r",
      "00:00:00     11 MB(0%)  Iter   1   81.82%  Align node\r",
      "00:00:00     11 MB(0%)  Iter   1   90.91%  Align node\r",
      "00:00:00     12 MB(0%)  Iter   1  100.00%  Align node\r",
      "00:00:00     12 MB(0%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:00     12 MB(0%)  Iter   1    8.33%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   16.67%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   25.00%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   33.33%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   41.67%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   50.00%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   58.33%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   66.67%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   75.00%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   83.33%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1   91.67%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "00:00:00     12 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00     12 MB(0%)  Iter   2   10.00%  Refine tree   \r",
      "00:00:00     13 MB(0%)  Iter   2   20.00%  Refine tree\r",
      "00:00:00     13 MB(0%)  Iter   2   30.00%  Refine tree\r",
      "00:00:00     13 MB(0%)  Iter   2  100.00%  Refine tree\r",
      "\r\n",
      "00:00:00     13 MB(0%)  Iter   2    8.33%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   16.67%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   25.00%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   33.33%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   41.67%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   50.00%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   58.33%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   66.67%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   75.00%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   83.33%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2   91.67%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "00:00:00     13 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00     13 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00     13 MB(0%)  Iter   3    9.52%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   14.29%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   19.05%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   23.81%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   28.57%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   33.33%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   38.10%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   42.86%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   47.62%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   52.38%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   57.14%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   61.90%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   66.67%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   71.43%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   76.19%  Refine biparts\r",
      "00:00:00     13 MB(0%)  Iter   3   80.95%  Refine biparts\r",
      "00:00:01     13 MB(0%)  Iter   3   85.71%  Refine biparts\r",
      "00:00:01     13 MB(0%)  Iter   3   90.48%  Refine biparts\r",
      "00:00:01     13 MB(0%)  Iter   3   95.24%  Refine biparts\r",
      "00:00:01     13 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "00:00:01     13 MB(0%)  Iter   3  104.76%  Refine biparts\r",
      "00:00:01     13 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "\r\n",
      "\r\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\r\n",
      "\r\n",
      "http://www.drive5.com/muscle\r\n",
      "This software is donated to the public domain.\r\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\r\n",
      "\r\n",
      "gene5gene 12 seqs, lengths min 608, max 609, avg 608\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\r",
      "\r\n",
      "00:00:00      1 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\r",
      "00:00:00      1 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\r",
      "\r\n",
      "00:00:00      2 MB(0%)  Iter   1    9.09%  Align node       \r",
      "00:00:00      3 MB(0%)  Iter   1   18.18%  Align node\r",
      "00:00:00      4 MB(0%)  Iter   1   27.27%  Align node\r",
      "00:00:00      4 MB(0%)  Iter   1   36.36%  Align node\r",
      "00:00:00      5 MB(0%)  Iter   1   45.45%  Align node\r",
      "00:00:00      5 MB(0%)  Iter   1   54.55%  Align node\r",
      "00:00:00      5 MB(0%)  Iter   1   63.64%  Align node\r",
      "00:00:00      6 MB(0%)  Iter   1   72.73%  Align node\r",
      "00:00:00      6 MB(0%)  Iter   1   81.82%  Align node\r",
      "00:00:00      6 MB(0%)  Iter   1   90.91%  Align node\r",
      "00:00:00      7 MB(0%)  Iter   1  100.00%  Align node\r",
      "00:00:00      7 MB(0%)  Iter   1  100.00%  Align node\r",
      "\r\n",
      "00:00:00      7 MB(0%)  Iter   1    8.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   16.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   25.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   33.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   41.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   50.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   58.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   66.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   75.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   83.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1   91.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   1  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00      7 MB(0%)  Iter   2   10.00%  Refine tree   \r",
      "00:00:00      7 MB(0%)  Iter   2   20.00%  Refine tree\r",
      "00:00:00      7 MB(0%)  Iter   2   30.00%  Refine tree\r",
      "00:00:00      7 MB(0%)  Iter   2   40.00%  Refine tree\r",
      "00:00:00      7 MB(0%)  Iter   2   50.00%  Refine tree\r",
      "00:00:00      7 MB(0%)  Iter   2   60.00%  Refine tree\r",
      "00:00:00      7 MB(0%)  Iter   2  100.00%  Refine tree\r",
      "\r\n",
      "00:00:00      7 MB(0%)  Iter   2    8.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   16.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   25.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   33.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   41.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   50.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   58.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   66.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   75.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   83.33%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2   91.67%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "00:00:00      7 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00      7 MB(0%)  Iter   2  100.00%  Root alignment\r",
      "\r\n",
      "00:00:00      7 MB(0%)  Iter   3    9.52%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   14.29%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   19.05%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   23.81%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   28.57%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   33.33%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   38.10%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   42.86%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   47.62%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   52.38%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   57.14%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   61.90%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   66.67%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   71.43%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   76.19%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   80.95%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   85.71%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   90.48%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3   95.24%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3  104.76%  Refine biparts\r",
      "00:00:00      7 MB(0%)  Iter   3  100.00%  Refine biparts\r",
      "\r\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for file in *gene.fasta\n",
    "do \n",
    "out=${file%%gene.fasta}aln.fasta\n",
    "muscle -in $file -out $out\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wrote concatenated sequences to fasta file 'concatenated.out'\r\n",
      "Wrote partitions for the concatenated file to 'partitions.txt'\r\n",
      "\n"
     ]
    }
   ],
   "source": [
    "python3 AMAS.py concat -f fasta -d dna -i *aln.fasta --part-format raxml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     ""
    ],
    [
     "Python3",
     "python3",
     "Python3",
     "#FFD91A",
     {
      "name": "ipython",
      "version": 3
     }
    ]
   ],
   "panel": {
    "displayed": true,
    "height": 0
   },
   "version": "0.21.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
